### PR TITLE
require specific versions of sphinxcontrib to gen docs

### PR DESCRIPTION
**Pull request by bratpiorka**
this PR fixes building HTML docs

### CODE REVIEW

**Review by ldorau — APPROVED** [review]

**Comment by PatKamin** [comment]
Please, provide a link to the html docs preview.

**Review by lukaszstolarczuk — COMMENTED** [review]
Do you know why did it stop working exactly...? It looks like it was out of the blue, after Igor's change...

My concern is, that bumping Sphinx version may lead to stop working for other OSes (like Rocky Linux or OpenSUSE - that was my experience with dockers on UR). I'm not sure how much you care about it 😉

File: third_party/requirements.txt [hunk 1]
```diff
@@ -2,6 +2,6 @@
 # Formatting the source code
 clang-format==15.0.7
 # Generating HTML documentation
-breathe==4.35.0
-sphinx==4.5.0
-sphinx_book_theme==0.3.3
+breathe>=4.35.0
+sphinx>=5.0.0
+sphinx_book_theme>=0.3.3
```
Comment (lukaszstolarczuk) [inline third_party/requirements.txt:7]: Please, update the commit message with some details why do we need this change.

**Comment by bratpiorka** [comment]
> Please, provide a link to the html docs preview.

@PatKamin when I require specific versions of sphinx contrib I fix the building while the layout doesn't change, see: https://bratpiorka.github.io/unified-memory-framework/api.html#memory-provider

File: third_party/requirements.txt (hunk 1 above)
Comment (bratpiorka) [reply third_party/requirements.txt:7]: I changed my PR because I wanted to keep the current layout

**Comment by bratpiorka** [comment]
> Do you know why did it stop working exactly...? It looks like it was out of the blue, after Igor's change...
> 
> My concern is, that bumping Sphinx version may lead to stop working for other OSes (like Rocky Linux or OpenSUSE - that was my experience with dockers on UR). I'm not sure how much you care about it 😉
.
@lukaszstolarczuk I'm not sure... the only thing I saw is the different version of github image provisioner. Maybe Python requirements were searched and installed in a different order? Anyway, I kept the current version of Sphinx

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**